### ISSUE TITLE
Ver 1.5.5a hangs, does nothing

### ISSUE THREAD

**Issue by Angel996** [nicehash/excavator#266]
DL'ded the 1.5.5a today (vs. 1.3.0 used previously), here is the output:

```
sudo /usr/bin/excavator -u 3xxxxxxxxxxxxxxxxxxxxxxxxxxxxx.RIG05 -a equihash
[sudo] password for user:

=========================== www.nicehash.com =========================
           Excavator v1.5.5a_nvidia GPU Miner for NiceHash.
           Copyright (C) 2018 NiceHash. All rights reserved.
=========================== www.nicehash.com =========================

Build time: 2018-06-14 14:32:32+02:00
Build number: 3287280602
[11:16:13][0x00007f9b6932f740][info] Log started
[11:16:13][0x00007f9b6932f740][info] core | Found CUDA device: GeForce GTX 1060 3GB
[11:16:13][0x00007f9b6932f740][info] core | Found CUDA device: GeForce GTX 1060 3GB
[11:16:13][0x00007f9b6932f740][info] core | Found CUDA device: GeForce GTX 1060 3GB
[11:16:13][0x00007f9b6932f740][info] core | Found CUDA device: GeForce GTX 1060 3GB
[11:16:13][0x00007f9b6932f740][info] core | Found CUDA device: GeForce GTX 1060 3GB
[11:16:13][0x00007f9b6932f740][info] api | Listening on 127.0.0.1:3456
[11:16:13][0x00007f9b6932f740][info] core | Initialized!
```

Tried equihash and lyra2rev2. Just nothing happens. The rig runs fine other miners (ethminer, dstm, bminer etc). Ubuntu 16.04 LTS.

**Comment by dropky** [comment]
From version 1.4.0 up excavator doesn't support command line parameters for algorithms and user anymore. You have to run excavator using config file: `excavator -c path_to_config_file.json`

Here is an example of config file to run equihash on first GPU: https://github.com/nicehash/excavator/blob/master/default_command_file.json

To run equihash on all 4 GPUs, your config file should look like this (if you want to run lyra2rev2, just change parameters in algorithm.add and worker.add methods from equihash to lyra2rev2):
```
[
	{"time":0,"commands":[
		{"id":1,"method":"subscribe","params":["nhmp.usa.nicehash.com:3200","btc-addr.workerName"]}
	]},
	{"time":2,"commands":[
       		{"id":1,"method":"algorithm.add","params":["equihash"]}
        ]},
	{"time":3,"commands":[
		{"id":1,"method":"worker.add","params":["equihash","0"]},
		{"id":1,"method":"worker.add","params":["equihash","1"]},
		{"id":1,"method":"worker.add","params":["equihash","2"]},
		{"id":1,"method":"worker.add","params":["equihash","3"]}
	]},
	{"time":10,"loop":10,"commands":[
		{"id":1,"method":"worker.print.speeds","params":[]},
		{"id":1,"method":"algorithm.print.speeds","params":[]}
	]}
]
```

Please take a look at API documentation: https://github.com/nicehash/excavator/tree/master/api#methods

**Comment by Angel996** [comment]
Thanks for the info.

Maybe it would be a good idea to display some message to user in this case?

Whats nhmp subdomain on Nicehash? Auto-switch pool? I can't find any info on that on NH website. Thanks!

**Comment by dropky** [comment]
> Maybe it would be a good idea to display some message to user in this case?

You're right, we'll add a cmd message in case of incorrect startup parameters.

> Whats nhmp subdomain on Nicehash? Auto-switch pool? 


From version 1.5.0 up we implemented our own stratum protocol - nhmp (nicehash miner protocol). Stratum servers that support nhmp protocol are located at nhmp.LOCATION.nicehash.com:3200. We're using the same port for all algorithms. It's similar to old stratum addresses which were algorithm specific (e.g equihash.LOCATION.nicehash.com:3357).

**Comment by Angel996** [comment]
Ok, still nothing happens.

sudo excavator -c ./excavator-lyra2v2.json

_Build time: 2017-11-21 16:44:49+00:00
Build number: 1
[18:44:15][0x00007f59e7235740][info] Log started
[18:44:15][0x00007f59e7235740][info] core | Found CUDA device: GeForce GTX 1060 3GB
[18:44:15][0x00007f59e7235740][info] core | Found CUDA device: GeForce GTX 1060 3GB
[18:44:15][0x00007f59e7235740][info] core | Found CUDA device: GeForce GTX 1060 3GB
[18:44:15][0x00007f59e7235740][info] core | Found CUDA device: GeForce GTX 1060 3GB
[18:44:15][0x00007f59e7235740][info] core | Found CUDA device: GeForce GTX 1060 3GB
[18:44:15][0x00007f59e7235740][info] core | Found CUDA device: GeForce GTX 1060 3GB
[18:44:15][0x00007f59e7235740][info] api | Listening on 127.0.0.1:3456
[18:44:15][0x00007f59e7235740][info] core | Initialized!
[18:44:18][0x00007f59e1862700][warning] core | Command exec: stoi_

Contents of json file:

[
    {"time":0,"commands":[
        {"id":1,"method":"subscribe","params":["nhmp.eu.nicehash.com:3200","3P**********************************ux.RIG04"]}
    ]},
    {"time":2,"commands":[
        {"id":1,"method":"algorithm.add","params":["lyra2rev2"]}
        ]},
    {"time":3,"commands":[
        {"id":1,"method":"worker.add","params":["lyra2rev2","0"]},
        {"id":1,"method":"worker.add","params":["lyra2rev2","1"]},
        {"id":1,"method":"worker.add","params":["lyra2rev2","2"]},
        {"id":1,"method":"worker.add","params":["lyra2rev2","3"]}
    ]},
    {"time":10,"loop":10,"commands":[
        {"id":1,"method":"worker.print.speeds","params":[]},
        {"id":1,"method":"algorithm.print.speeds","params":[]}
    ]}
]

**Comment by dropky** [comment]
Hm, this should work just fine. Can you run excavator with -d 0 flag and c/p the cmd output here (`excavator -c excavator-lyra2v2.json -d 0`) ?

**Comment by Angel996** [comment]
Got it working on the 6x 1060 rig, thanks!
